src/components/DetailsPage/ProductRoute.js

```diff
@@ -0,0 +1,26 @@
+import { useParams } from 'react-router-dom'
+import DetailsPage from './DetailsPage'
+
+function ProductRoute({ productData }) {
+  const { id } = useParams()
+
+  const productInfo = productData.find(product => product.id === id)
+
+  return productInfo ? (
+    <DetailsPage
+      name={productInfo.name}
+      image={productInfo.image}
+      brand={productInfo.brand}
+      description={productInfo.description}
+      ingredients={productInfo.ingredients}
+      nutritionFacts={productInfo.nutritionFacts}
+      stores={productInfo.stores}
+      categories={productInfo.categories}
+      price={productInfo.price}
+    />
```

Comment:
This one took me a moment:
I looked into your vercel, because of the Link in the DetailsPage. In vercel it was `undefined` and I got curious.
You dont have the id in your props that you are giving down to DetailsPage

```
<DetailsPage
      id={id}
      name={productInfo.name}
      image={productInfo.image}
      brand={productInfo.brand}
      description={productInfo.description}
      ingredients={productInfo.ingredients}
      nutritionFacts={productInfo.nutritionFacts}
      stores={productInfo.stores}
      categories={productInfo.categories}
      price={productInfo.price}
    />
```

Tbh, I think this adding the DetailsPage into an own component is not really necessary(, yet). It probably makes more sense to have the DetailsPage already here and then break it into smaller components which then show the Details.

Reply:
Thanks for your effort reviewing this! You're right, if I want to have a link with the id on the DetailsPage, I need to make sure it has access to the id.
I removed the link in the DetailsPage.js, so I don't need the id as a prop for now.